src/components/general/ApplicationGuidance/components/Anchor.tsx

```diff
@@ -1,54 +1,85 @@
 import React from 'react';
 import {
-    AppGuideAnchorElement,
-    AppGuideAnchorElementProps,
-    QuickFactToggleEventData,
-} from '../../../../customElements/components/app-guide/anchor';
+    AnchorDOMRect,
+    OverlayAnchorElement,
+    OverlayAnchorElementTag,
+    OverlayAnchorElementProps,
+    OverlayAnchorConnectEvent
+} from '../../../../customElements/components/overlay/anchor';
 
-export type ApplicationGuidanceAnchorProps = React.PropsWithChildren<AppGuideAnchorElementProps>;
+export interface AppGuideAnchorRef<R extends HTMLElement> {
+    id: string;
+    scope: string;
+    padding?: number;
+    ref: React.RefObject<R>;
+}
+
+export const useAnchor = <R extends HTMLElement>(anchor: Omit<AppGuideAnchorRef<R>, 'ref'>) => {
```

Comment:
Maybe add some documentation/comments for this. The hook name "useAnchor" is not very self explanatory